### PR TITLE
Add support for type annotations (#254)

### PR DESCRIPTION
**Pull request by pgrandjean**

### CODE REVIEW

**Comment by pgrandjean** [comment]
I think this PR is ready for review

**Comment by propensive** [comment]
Thanks @pgrandjean! Sorry I haven't had an opportunity to look at it in detail yet. If @joroKr21 can get a chance to review it, I'm happy to merge it, otherwise, I will hopefully get a chance later in the week.

**Comment by joroKr21** [comment]
Sorry, It’s been a busy month at work. I think in one or two weeks I will have more time for OSS.

**Comment by propensive** [comment]
I may still get there first, so hopefully we're not holding you up too much, @pgrandjean.

**Comment by pgrandjean** [comment]
No problem!

**Comment by pgrandjean** [comment]
Hi there! Would you have any news on a possible review?

File: core/shared/src/main/scala/magnolia.scala [hunk 1]
```diff
@@ -175,9 +175,32 @@ object Magnolia {
     }
 
     def annotationsOf(symbol: Symbol): List[Tree] = symbol.annotations.collect {
-      case annotation if !(annotation.tree.tpe <:< javaAnnotationType) => annotation.tree
+      case annotation if !(annotation.tree.tpe <:< javaAnnotationType) =>
+        annotation.tree
```
Comment (joroKr21) [inline core/shared/src/main/scala/magnolia.scala:179]: We should create a helper method out of this `collect` and use it below as well.

File: core/shared/src/main/scala/magnolia.scala [hunk 2]
```diff
@@ -175,9 +175,32 @@ object Magnolia {
...
     }
 
+    def typeAnnotationsOf(symbol: Symbol, fromParents: Boolean): List[Tree] = {
+      val tpeAnns = if (fromParents) {
+        val parents = symbol.typeSignature match {
+          case ClassInfoType(ps, _, _) => ps.filterNot(_ == symbol.typeSignature)
```
Comment (joroKr21) [inline core/shared/src/main/scala/magnolia.scala:185]: Why is this necessary?

File: core/shared/src/main/scala/magnolia.scala (hunk 1 above)
Comment (joroKr21) [reply core/shared/src/main/scala/magnolia.scala]: I pushed a commit to do that

File: core/shared/src/main/scala/magnolia.scala (hunk 2 above)
Comment (joroKr21) [reply core/shared/src/main/scala/magnolia.scala]: Removed the filter - it was not necessary

**Comment by joroKr21** [comment]
Thanks @pgrandjean for your contribution and patience 😄

**Comment by pgrandjean** [comment]
Sorry for the late reply. Thank you very much for the merge!

**Comment by pgrandjean** [comment]
@joroKr21 Do you mind if I ask when the next release will be?

**Comment by joroKr21** [comment]
That's up to @propensive

**Comment by propensive** [comment]
I just triggered the publish procedure. Keep an eye on [Travis](https://travis-ci.org/github/propensive/magnolia/builds/718409532).

**Comment by pgrandjean** [comment]
@propensive thank you!
